src/bindings/python/CMakeLists.txt

```diff
@@ -115,6 +115,8 @@ set(INIT_FILES_RUNTIME "${OpenVINOPython_SOURCE_DIR}/src/openvino/__init__.py"
 ov_check_init_files_alignment("${INIT_FILES_RUNTIME}")
 
 ov_option(ENABLE_PYTHON "Enables OpenVINO Python API build" ${ENABLE_PYTHON_DEFAULT})
+# we should not cache the option to avoid situations when ENABLE_PYTHON is ON but ENABLE_PYTHON_DEFAULT is OFF
+unset(ENABLE_PYTHON CACHE)
```

Comment:
Why situations when ENABLE_PYTHON is ON but ENABLE_PYTHON_DEFAULT is OFF must be avoided. It seemed to be a valid stated to me.

Reply:
the case is simple, we could not detected Python and set ENABLE_PYTHON_DEFAULT to OFF, then ENABLE_PYTHON should be also set to OFF, but we will take cached value (ON)

Reply:
decided that it the right behavior, user would need to set this option manually or reset the cache